### PR TITLE
64 bit simulation support

### PR DESCRIPTION
**Pull request by natestuder**
Small changes needed to enable 64-bit support for bare-metal.

### CODE REVIEW

File: freedom-e-sdk/bsp/env/start.S [hunk 1]
```diff
@@ -14,28 +15,33 @@ _start:
 	.cfi_undefined ra
 .option push
 .option norelax
-	//la gp, __global_pointer$
-	lui gp, %hi(__global_pointer$)
-	addi gp, gp, %lo(__global_pointer$)
+1:
+   //la gp, _global_pointer$
+   auipc gp, %pcrel_hi(__global_pointer$)
+	addi gp, gp, %pcrel_lo(1b)
```
Comment (JohnPMerrill) [inline freedom-e-sdk/bsp/env/start.S]: Remove indentation (maybe github is doing this)

File: freedom-e-sdk/bsp/libwrap/stdlib/malloc.c [hunk 2]
```diff
@@ -28,11 +28,17 @@
 #include <stdlib.h>
 #include <stdint.h>
 
-#define configTOTAL_HEAP_SIZE	    ( ( size_t ) ( 0x1000 ) )
+#define configTOTAL_HEAP_SIZE	    ( ( size_t ) ( 0x2000 ) )
 
+#if __riscv_xlen == 64
+#define portPOINTER_SIZE_TYPE uint64_t
```
Comment (JohnPMerrill) [inline freedom-e-sdk/bsp/libwrap/stdlib/malloc.c]: `uintptr_t`

File: freedom-e-sdk/bsp/libwrap/stdlib/malloc.c [hunk 3]
```diff
@@ -28,11 +28,17 @@
 #include <stdlib.h>
 #include <stdint.h>
 
-#define configTOTAL_HEAP_SIZE	    ( ( size_t ) ( 0x1000 ) )
+#define configTOTAL_HEAP_SIZE	    ( ( size_t ) ( 0x2000 ) )
 
+#if __riscv_xlen == 64
+#define portPOINTER_SIZE_TYPE uint64_t
+#define portBYTE_ALIGNMENT 8
```
Comment (JohnPMerrill) [inline freedom-e-sdk/bsp/libwrap/stdlib/malloc.c]: `(__riscv_xlen / 4)`

File: freedom-e-sdk/bsp/env/start.S (hunk 1 above)
Comment (natestuder) [reply freedom-e-sdk/bsp/env/start.S]: Mixed tabs and spaces...

File: freedom-e-sdk/bsp/libwrap/stdlib/malloc.c (hunk 3 above)
Comment (natestuder) [reply freedom-e-sdk/bsp/libwrap/stdlib/malloc.c]: `__riscv_xlen / 8`

Comment (JohnPMerrill) [reply freedom-e-sdk/bsp/libwrap/stdlib/malloc.c]: gotta brush up on my multiplication tables...

**Pull request merged**